include/mbedtls/platform_util.h

```diff
@@ -300,6 +300,15 @@ uint32_t mbedtls_platform_random_in_range( uint32_t num );
  */
 void mbedtls_platform_random_delay( void );
 
+/**
+ * \brief       RNG-function for getting a random buffer.
+ *
+ * \param buf   Buffer for random data
+ * \param len   Length of the buffer in bytes
+ *
+ */
+void mbedtls_platform_random_buf( uint8_t *buf, size_t len);
```

Comment:
I'm not sure if extending the API to get a buffer is needed. We already have an API for getting uint16_t and uint32_t, users can take as much data as they want on their side. This will increase the library size for each place it's used in, even if the users don't want a buffer, just smaller data amounts.

Reply:
@AndrzejKurek uint16 API is our internal one, it exists because of Alexander recommendation not use all 32 bites of generated random, and create 32 bit using mix of 2 different generation.
I think that providing possibility to generate  unlimited number of bytes is important, otherwise all users will have to implement the same loop by their-self.